src/mpid/ch4/src/ch4_impl.h

```diff
@@ -1146,4 +1147,40 @@ MPL_STATIC_INLINE_PREFIX MPI_Op MPIDIU_win_acc_get_op(int index)
         return MPIR_Op_builtin_get_op(index);
     }
 }
+
+MPL_STATIC_INLINE_PREFIX bool MPIDIG_rma_recv_am_always_poll(void)
+{
+    bool always_poll = false;
+    int interval;
+
+    MPIR_cc_incr(&MPIDIG_global.rma_recv_am_poll_cntr, &interval);
+
+    if (MPIR_CVAR_CH4_RMA_RECV_AM_PROGRESS_INTERVAL > 0) {
+        /* User explicitly controls the polling frequency */
+        always_poll =
+            ((interval + 1) % MPIR_CVAR_CH4_RMA_RECV_AM_PROGRESS_INTERVAL == 0) ? true : false;
+    } else if (MPIR_CVAR_CH4_RMA_RECV_AM_PROGRESS_INTERVAL == -1) {
+        /* Runtime control is on.
+         * - Always poll if RMA receive AM is active
+         * - Otherwise poll with longer interval */
+        if (MPL_atomic_load_int(&MPIDIG_global.rma_recv_am_active)) {
+            always_poll = true;
+        } else {
+            always_poll =
+                ((interval + 1) % MPIR_CVAR_CH4_RMA_RECV_AM_PROGRESS_INTERVAL_LONG ==
+                 0) ? true : false;
+        }
+    } else {
+        /* User explicitly disables polling */
+        always_poll = false;
+    }
+
+    return always_poll;
+}
+
+MPL_STATIC_INLINE_PREFIX void MPIDIG_rma_recv_am_activate(void)
```

Comment:
This name is another sticking point. It is not "activate" "recv_am", right? It is merely updating/setting a flag indicating we have active messages targeting this process. So I would like to see the semantics of "updating/setting flag" intuitive from the name.

Reply:
I tried to follow the activate term used to enable/disable progress hooks. But yes, we don't really activate recv_am. How about `MPIDIG_rma_set_target_am_arrived`?

Reply:
I like the `MPIDIG_rma_set_` part. "target_am" is very confusing. "arrived" is also not exactly the intention. The intention is we don't care about "arrived" am message, but having received am message indicates that we may receive future unexpected am messages. It is too bad you don't like `unexp`, I think it really describes exactly your intention. @raffenet what is your opinion?

Reply:
Similar to the other case, I don't think we need to state "why" we're updating the am flag in the API. That is determined at the call site. Would this API be used if the origin process issues its own active message RMA? Or are we explicitly distinguishing when active messages have been issued to this process (target)?

Reply:
My intent is to explicitly distinguish when active messages have been issued to this process (target). Because we never reduce the polling frequency on the origin process when the origin process issues its own active message RMA.